benches/basic.rs

```diff
@@ -126,3 +133,75 @@ bench_over_inputs!(eq, |b, input| {
     let second = black_box(first.clone());
     b.iter(|| first == second);
 });
+
+macro_rules! contended_from_str {
+    ($([$name:ident, $thread_count:expr];)+) => {
+        criterion::criterion_group!(
+            multithreaded_contended,
+            $($name),+
+        );
+
+        $(
+            bench_over_inputs!($name, |b, input| {
+                std::thread::scope(|s| {
+                    b.iter(|| {
+                        let handles = std::array::from_fn::<_, $thread_count, _>(|_| {
+                            s.spawn(|| FlyStr::from(input))
+                        });
+
+                        handles.map(|handle| handle.join().unwrap())
```

Comment:
I tried running this locally and it seems like the runtime is dominated by the thread lifecycle. I wonder if you could measure the string creation without spawning or joining the threads inside `b.iter()`?